### PR TITLE
Introduce `allow_negative` flag in tensor check

### LINKED ISSUE #168: Reasons for assert torch.all(tensor >= 0)
Inside the _validate_input function, there is a check for values of the input tensor to be >=0: https://github.com/photosynthesis-team/piq/blob/master/piq/utils/common.py#L46

However, this introduces difficulties when using some of the losses from piq. For example, I would like to use DISTS as a loss function for training a neural network without an activation function at the end. 

Could you please clarify the reasons why this check is required?

### PR DESCRIPTION
**Pull request by zakajd**
Closes #168 and relevant to #133 

## Proposed Changes
  - Flag `allow_negative`
  - Minor documentation update

### CODE REVIEW

**Comment by snk4tr** [comment]
@zakajd conflicts need to be resolved.

**Comment by zakajd** [comment]
@snk4tr  ready to be merged

**Review by snk4tr — APPROVED** [review]
